zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -120,29 +131,179 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
         } else if ("endDate".equals(field)) {
           builder.endDate();
         } else {
-          throw new RuntimeException("unkown sortBy " + field);
+          // Collect validation error for unknown sortBy field
+          validationErrors.add(ProblemDetail.forStatusAndDetail(
+              HttpStatus.BAD_REQUEST, "Unknown sortBy field: " + field));
+          continue;
         }
 
         if ("asc".equalsIgnoreCase(order)) {
           builder.asc();
         } else if ("desc".equalsIgnoreCase(order)) {
           builder.desc();
         } else {
-          throw new RuntimeException("unkown sortOrder " + order);
+          // Collect validation error for unknown sortOrder
+          validationErrors.add(ProblemDetail.forStatusAndDetail(
+              HttpStatus.BAD_REQUEST, "Unknown sortOrder: " + order));
         }
       }
 
+      if (!validationErrors.isEmpty()) {
+        return Either.left(validationErrors.get(0)); // Return the first encountered error for simplicity
+      }
+
       return Either.right(builder.build());
     }
 
     return Either.right(null);
   }
 
+  public static Either<ProblemDetail, UserTaskSort> toUserTaskSearchQuerySort(
+      final List<SearchQuerySortRequest> sorting) {
+
+    if (sorting != null && !sorting.isEmpty()) {
+      final var builder = SortOptionBuilders.userTask();
+      final List<ProblemDetail> validationErrors = new ArrayList<>();
+
+      for (final SearchQuerySortRequest sort : sorting) {
+        final var field = sort.getField();
+        final var order = sort.getOrder();
+
+        if ("creationTime".equals(field)) {
+          builder.creationDate();
+        } else if ("completionTime".equals(field)) {
+          builder.completionDate();
+        } else {
+          // Collect validation error for unknown sortBy field
+          validationErrors.add(ProblemDetail.forStatusAndDetail(
+              HttpStatus.BAD_REQUEST, "Unknown sortBy field: " + field));
+          continue;
+        }
+
+        if ("asc".equalsIgnoreCase(order)) {
+          builder.asc();
+        } else if ("desc".equalsIgnoreCase(order)) {
+          builder.desc();
+        } else {
+          // Collect validation error for unknown sortOrder
+          validationErrors.add(ProblemDetail.forStatusAndDetail(
+              HttpStatus.BAD_REQUEST, "Unknown sortOrder: " + order));
+        }
+      }
+
+      if (!validationErrors.isEmpty()) {
+        return Either.left(validationErrors.get(0)); // Return the first encountered error for simplicity
```

Comment:
❌ Same as above, collecting all errors into one response is more helpful to users.